### DISCUSSION TITLE
How Compare (as matched) 2 files but one with four digit numeric prefix

### DISCUSSION THREAD

**Discussion by sagegriesfox** [WinMerge/winmerge#1301] [General]
I have two folders of mp3 music files, but a disk corruption event caused the most current folder to be ~50% destroyed.  Each of those current folder files was cleaned up over a 3-month extended period, volume levels fixed, and 2 sec pre/post silences added  at the beginning and end of files.

I'm now trying to determine which of the complete collection of tunes in the older (uncorrected) folder files with no 4-digit sequence # prefixed to the file titles are missing from the new folder (corrected) files with prefix number.  So I can get them and clean them up and transfer them into the new folder.

Because of the audio conforming efforts on the new files  the file binary contents are likely to be different, so the only comparison means is by titles.

I have not been able to design a Filter RegEdit sequence to allow comparing ...  "**1234-Maria**" to "**Maria**" (which are "matched" files).

I tried this but it doesn't work....
   f: \d+-(.+) ## Filter for filename
  d: \\subdir$ ## Filter for directory  (I don't need subdirectory as the folder is all files, but didn't know if I needed to remove the "d:" 
       entry or not).

Bottom line.  Can I filter the file titles to remove the prefix sequence numbering so the titles match between the two folders.
If the answer is "yes" then what filter string should I be using?

Thanks.

**Comment by sdottaka** [comment]
Unfortunately, file filters are meant to exclude specified files or folders from comparisons and cannot be used to match files with different filenames.

**Comment by KerimF1** [comment]
[KerimF1](https://github.com/KerimF1)
[4 minutes ago](https://github.com/WinMerge/winmerge/discussions/1301#discussioncomment-2622528)
If I need solving this case (on W7) I let Excel do the comparison (after filtering) of the two dir lists which could be created by the DOS DIR command.
In each of the two folders of interest I run a 1-line bat file (named dir.bat for example):
`dir *.* >dir.txt`
or
`dir *.* /S >dir.txt` (to include subdirectories)

**Comment by sagegriesfox** [comment]
If it isn't with Filtering, can I somehow match almost same files but with
different names (one set of files has a 2-sec silence appended to the
beginning/end of file, and the other set has random silence periods).

Some file Difference Apps can do that but don't offer useful post scan
options to do anything with the results like you do.

I have 7000 files to compare and need as much automation as possible (as
you provide).

You are not matching at all any files which are slightly different in mane
and loudness or tiny silence period differernces.

I was hoping you would have what I need, but I need to match, or not, mp3
audio files with slightly different names and slightly different 5% ?
content differences.

Is this possible with WinMerge?

Thanks,
 ken b

**Comment by sdottaka** [comment]
Unfortunately, WinMerge does not have the ability to compare files with different file names in a folder comparison.

In order to compare files with WinMerge, you will need to create symbolic links so that the file names match.
